File: daily.go
package main

import (
	"fmt"
)

func minPatches(nums []int, n int) int {
	total := 0
	count := 0
	index := 0
	for total < n {
		if index < len(nums) && nums[index] <= total+1 {
			total += nums[index]
			index++
		} else {
			total = total + (total + 1)
			count++
		}
	}
	return count
}
func main() {
	nums := []int{1, 3}
	n := 6
	nums = []int{1, 5, 10}
	n = 20
	fmt.Println(minPatches(nums, n))
}

File: daily.rs
impl Solution {
    pub fn min_patches(nums: Vec<i32>, n: i32) -> i32 {
        let (mut max,mut cnt)=(1,0); //exclusive
        nums.into_iter().filter(|&x|x<=n).for_each(|x|if max>=x as i64{max+=x as i64}else{while x as i64>max{max<<=1;cnt+=1}max+=x as i64});
        while n as i64>=max{max<<=1;cnt+=1}
        cnt
    }
}